biography/models.py

```diff
@@ -1,125 +1,114 @@
-        help_text=_("context and description of the dump"))
-    notes = models.TextField(
-        _("Notes"), blank=True,
-        help_text=_("notes by and for the biographer"))
-    created = models.DateTimeField(auto_now_add=True)
-    modified = models.DateTimeField(auto_now=True)
-
-    class Meta:
-        verbose_name = _("Memoir")
-        verbose_name_plural = _("Memoirs")
-
-    def get_auto_memoir(self):
-        return self.objects.get(author__is_null=True)
+class User(AbstractUser):
+    is_editor = models.BooleanField(default=False)
+    is_person = models.BooleanField(default=False)
 
 
-class Period(models.Model):
-    ''' a meaningful period in a life '''
-    name = models.CharField(max_length=80)
-    story_audio = models.FileField(_("Story Audio"), null=True, blank=True)
-    story_text = models.TextField(_("Story Text"), blank=True)
-    notes = models.TextField(
-        _("Notes"), blank=True,
-        help_text=_("notes by and for the biographer"))
-    place = models.TextField(_("Place"))
-    start_date = models.DateField(_("Start Date"))
-    end_date = models.DateField(_("End Date"))
-    biography = models.ForeignKey(
-        'Biography', verbose_name=_("Biography"), on_delete=models.CASCADE)
-
-    gallery = models.ForeignKey(
-        Gallery, null=True, blank=True, on_delete=models.SET_NULL)
-
-    class Meta:
-        verbose_name = _("Period")
-        verbose_name_plural = _("Periods")
-
-
-class Biography(models.Model):
-    ''' a person's life story '''
-    uuid = models.UUIDField(
-        primary_key=True, default=uuid.uuid4, editable=False)
-    # TODO: dates should be fuzzy
+class Person(models.Model):  # Biography
+    ''' A person's life story'''
+    uuid = models.UUIDField(primary_key=True,
+                            default=uuid.uuid4,
+                            editable=False)
+    is_public = models.BooleanField(_("Public"),
+                                    default=False,
+                                    help_text=_('Is it public?'))
+    modified = models.DateTimeField(auto_now=True)
+    first_name = models.TextField()
+    last_name = models.TextField()
     date_of_birth = models.DateTimeField(
         _("Date of birth"), blank=True, null=True)
     date_of_passing = models.DateTimeField(
         _("Date of passing"), blank=True, null=True)
     place_of_birth = models.TextField(_("Place of birth"))
     place_of_passing = models.TextField(
         _("Place of passing"), blank=True, null=True)
-    user = models.ForeignKey(
-        User, verbose_name=_("Person"), on_delete=models.CASCADE,
-        help_text=_('all other personal data is stored on the user'))
-    editors = models.ManyToManyField(User, verbose_name=_("Editors"),
-        related_name='editing_bios',
-        help_text=_('the good people that can edit this biography'))
-    # creator is critical when we have issues of ownership and liability
+    editors = models.ManyToManyField(User,
```

Comment:
Most system have also viewer and owner. We will need the first for private biographies and the latter for adding users, etc. Better use a through table.

Reply:
I figured this could be resolved by setting permissions by Groups (and/or `is_staff`). But I'll take another look.